File: js/algos/sorting/bubble_sort.js
function bubbleSort(arr){
	let swapped = false;
	for(let i=arr.length;i >0;i--){
		for(let j=0;j<i-1;j++){
			if(arr[j] > arr[j+1]){
				let tmp = arr[j];
				arr[j] = arr[j+1]
				arr[j+1] = tmp;
				swapped = true;
			}
		}
		if(!swapped) break;
	}
}

arr = [3,2,1,6,7,9];
bubble_sort(arr);
console.log(arr);

File: python/algos/sorting/bubble_sort.py
def bubble_sort(array):
	length = len(array)
	swapped = False
	for i in range(length,0,-1):
		for j in range(0,i):
			if array[j] > array[j+1]:
				array[j], array[j+1] = array[j+1], array[j]
				swapped = True

		if not swapped: break

arr = [3,2,1,6,9,7]
bubble_sort(arr)
print(arr)